clients/drcachesim/tools/view.cpp

```diff
@@ -205,6 +157,78 @@ view_t::process_memref(const memref_t &memref)
     return parallel_shard_memref(serial_stream_, memref);
 }
 
+bool
+view_t::init_decode_cache()
+{
+    assert(decode_cache_ == nullptr);
+    decode_cache_ =
+        std::unique_ptr<decode_cache_t<disasm_info_t>>(new decode_cache_t<disasm_info_t>(
+            dcontext_.dcontext,
+            // We will decode the instr_t ourselves.
+            /*include_decoded_instr=*/false,
+            /*persist_decoded_instrs=*/false, knob_verbose_));
+    if (TESTANY(OFFLINE_FILE_TYPE_ENCODINGS, filetype_)) {
+        error_string_ = decode_cache_->init(static_cast<offline_file_type_t>(filetype_));
+    } else if (!module_file_path_.empty()) {
+        error_string_ = decode_cache_->init(static_cast<offline_file_type_t>(filetype_),
+                                            module_file_path_, knob_alt_module_dir_);
+    } else {
+        // Continue but omit disassembly to support cases where binaries are
+        // not available and OFFLINE_FILE_TYPE_ENCODINGS is not present.
+        decode_cache_.reset(nullptr);
+    }
+    return error_string_.empty();
+}
+
+bool
+view_t::init_from_filetype()
+{
+    if (init_from_filetype_done_) {
+        return true;
+    }
+    if (filetype_ == OFFLINE_FILE_TYPE_DEFAULT) {
```

Comment:
Add a comment explaining why this is needed when there's a guard above.  Also, what if it really is the default?  What is this accomplishing?

Reply:
> Add a comment explaining why this is needed when there's a guard above.

You're right that it's not needed. Removed.

Reply:
> Also, what if it really is the default? What is this accomplishing?

You mean what if the trace's type really is OFFLINE_FILE_TYPE_DEFAULT? See other comment thread justifying use of OFFLINE_FILE_TYPE_DEFAULT to mean uninitialized.

Reply:
> You're right that it's not needed. Removed.

Sorry. This one's still needed because it's the fallback for when no filetype marker was found. Added the comment. Note the function is not init_filetype, it's init_from_filetype, meaning the initializations that need to happen using filetype.

Reply:
We may end up here despite filetype_ not being initialized yet. So we need this check.


The one above just prevents multiple init actions for decode cache etc